File: src/jk_5/nailed/map/gameloop/instructions/InstructionSetGamemode.java
package jk_5.nailed.map.gameloop.instructions;

import jk_5.nailed.Nailed;
import jk_5.nailed.map.gameloop.GameThread;
import jk_5.nailed.map.gameloop.IInstruction;
import jk_5.nailed.players.Player;
import jk_5.nailed.teams.Team;
import net.minecraft.src.EnumGameType;

/**
 * No description given
 *
 * @author jk-5
 */
public class InstructionSetGamemode implements IInstruction {

    private String team;
    private int gamemode;

    @Override
    public void injectArguments(String arguments) {
        String data[] = arguments.split(" ", 2);
        this.team = data[0];
        this.gamemode = Integer.parseInt(data[1]);
    }

    @Override
    public void execute(GameThread controller) throws InterruptedException {
        Team team = controller.getMap().getTeamManager().getTeam(this.team);
        for (Player p : Nailed.playerRegistry.getPlayers()) {
            if (p.getEntity() == null) continue;
            if (p.getTeam() == team) {
                p.getEntity().setGameType(EnumGameType.getByID(this.gamemode));
            }
        }
    }
}

File: src/main/scala/jk_5/nailed/map/gameloop/instructions/InstructionSetGamemode.scala
package jk_5.nailed.map.gameloop.instructions

import jk_5.nailed.map.gameloop.GameThread
import jk_5.nailed.map.gameloop.IInstruction
import net.minecraft.world.EnumGameType

/**
 * No description given
 *
 * @author jk-5
 */
class InstructionSetGamemode extends IInstruction {

  private var team: String = null
  private var gamemode: Int = 0

  def injectArguments(arguments: String) {
    val data = arguments.split(" ", 2)
    this.team = data(0)
    this.gamemode = Integer.parseInt(data(1))
  }

  def execute(controller: GameThread) {
    val team = controller.getMap.getTeamManager.getTeam(this.team)
    if(team.isEmpty) return
    controller.getMap.getPlayers.filter(p => p.getTeam.eq(team.get) && p.getEntity.isDefined).map(_.getEntity.get).foreach(_.setGameType(EnumGameType.getByID(this.gamemode)))
  }
}
